### PR TITLE
Until we support those, prevent text editing of JSON / list fields

### PR DESCRIPTION
**Pull request by nirvn**
QField does not (yet?) support JSON and list field types. ATM, it defaults to using the text editor widget, which displays a string representation of those objects in read view just fine, _but_ throws an empty edit field when editing features, which is set to a blank value and can't really be edited.

Until further notice, just show the string representation whether we're in edit mode or not. Showing that instead of a blank important edit field makes us look a lot better ;-P

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1035ace885a8f08cdc10f9b0669b12b9e72da7d4:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-1ad859f2ae27fa2c54432d278360e05146f7e811-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-1ad859f2ae27fa2c54432d278360e05146f7e811-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-1ad859f2ae27fa2c54432d278360e05146f7e811-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-1ad859f2ae27fa2c54432d278360e05146f7e811-x86.apk)

**Review by signedav — APPROVED** [review]
Cool

File: src/qml/editorwidgets/TextEdit.qml [hunk 1]
```diff
@@ -8,6 +8,8 @@ import "."
 EditorWidgetBase {
   id: topItem
 
+  property bool isEditable: isEnabled && typeof(value) != object
```
Comment (suricactus) [inline src/qml/editorwidgets/TextEdit.qml:11]:
```suggestion
  property bool isEditable: isEnabled && value && typeof(value) !== 'object'
```
note `typeof(null) === 'object'`

Have you tried this, the `object` part should be in quotes, no?

**Review by suricactus — CHANGES_REQUESTED** [review]
Does it work as expected? See my comment above.

**Comment by nirvn** [comment]
@suricactus , change committed.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b15bcb09272362bdcc9f63f9d1c6eca0518f0529:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-439e4e8ef02cac85cc2d8ddf9bb7abe7cdc76523-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-439e4e8ef02cac85cc2d8ddf9bb7abe7cdc76523-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-439e4e8ef02cac85cc2d8ddf9bb7abe7cdc76523-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1979-439e4e8ef02cac85cc2d8ddf9bb7abe7cdc76523-x86.apk)

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**